### PR TITLE
black 18.6b4 (new formula)

### PR DESCRIPTION
**Pull request by orn688**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Adds a formula for [Black](https://github.com/ambv/black), the uncompromising Python code formatter, addressing Black's issue [#362](https://github.com/ambv/black/issues/362).

### CODE REVIEW

File: Formula/black.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 "f15516df478d5a56180fbf80e68f206010e6d160fc39fa508b65e035fd75130b"
+  end
+
+  resource "toml" do
+    url "https://files.pythonhosted.org/packages/f5/f9/044110c267e6408013b85166a7cfcd352cf85275aa8ce700aa5c0eb407ba/toml-0.9.4.tar.gz"
+    sha256 "8e86bd6ce8cc11b9620cb637466453d94f5d57ad86f17e98a98d1f73e3baab2d"
+  end
+
+  def install
+    virtualenv_create(libexec, "python3")
```
Comment (JelleZijlstra) [inline Formula/black.rb]: Black requires at least Python 3.6, but I don't see anything in this file that prevents Homebrew from installing it under an earlier version of Python 3. Does something special need to be done?

I'm not familiar with Homebrew, so my apologies if this is not actually an issue.

Comment (orn688) [reply Formula/black.rb]: @JelleZijlstra That's a good point - I'm not a Homebrew expert and was unsure myself of whether this is an issue. After some research I couldn't find any way to ensure a certain version of a Homebrew formula dependency, but I finally stumbled upon the [Homebrew FAQ](https://docs.brew.sh/FAQ) (probably the first place I should've looked, but oh well) which states, on the subject of pinning packages:

> Note that pinned, outdated formulae that are depended on by another formula need to be upgraded when required as we do not allow formulae to be built against non-latest versions.

So long story short, installing Black via brew should automatically install and use Python 3.6 if it's not already installed.

File: Formula/black.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
+class Black < Formula
+  include Language::Python::Virtualenv
+
+  desc "The uncompromising Python code formatter"
```
Comment (fxcoudert) [inline Formula/black.rb]: `Python code formatter` is better: we want factual descriptions, not slogans :)

File: Formula/black.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/black.rb]: This will use Homebrew's python, which is currently 3.6.5 and will very soon be 3.7.0

**Review by fxcoudert — DISMISSED** [review]

File: Formula/black.rb (hunk 2 above)
Comment (orn688) [reply Formula/black.rb]: Fair point, I'll fix that!

File: Formula/black.rb (hunk 1 above)
Comment (orn688) [reply Formula/black.rb]: @fxcoudert Is this line necessary now that Homebrew's `python` is Python 3? According to the `Python for Formula Authors` docs you have to specify python3 when creating a virtualenv but it seems like that might be outdated.

Comment (fxcoudert) [reply Formula/black.rb]: Yes you need `python3` there

**Comment by orn688** [comment]
@fxcoudert Sorry to bother you, is there anything else I need to do before this can be merged?

File: Formula/black.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/black.rb]: the entire line is redundant and should be deleted.

**Comment by orn688** [comment]
@ilovezfs Thanks for the feedback, I removed the unnecessary line.

**Comment by hugovk** [comment]
@stale Awaiting review.

File: Formula/black.rb (hunk 2 above)
Comment (jacobwgillespie) [reply Formula/black.rb]: @orn688 it looks like this line was not changed.

Comment (orn688) [reply Formula/black.rb]: Oops! Not sure how I messed that up. Thanks for catching it!

**Comment by DomT4** [comment]
@BrewTestBot test this please

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/4dec1d2bc673322e03f9c158c582775fbcd52343. Thank you for your contribution to Homebrew @orn688; we appreciate it! :smiley_cat:
